tests/server_test.go

```diff
@@ -9857,6 +9858,111 @@ func TestServer_Prometheus_Write(t *testing.T) {
 	}
 }
 
+func TestCliEndToEnd(t *testing.T) {
+	config := NewConfig()
+	config.HTTPD.FluxEnabled = true
+	s := OpenServer(config)
+	defer s.Close()
+
+	// Some test data
+	s.CreateDatabase(t.Name())
+	defer s.DropDatabase(t.Name())
+	points, err := models.ParsePointsString(`
+	m0,k=k0 f=0i 0
+	m0,k=k0 f=1i 1
+	m0,k=k0 f=2i 2
+	m0,k=k0 f=3i 3
+	m0,k=k0 f=4i 4
+	m0,k=k1 f=5i 5
+	m0,k=k1 f=6i 6
+	m1,k=k0 f=5i 7
+	m1,k=k2 f=0i 8
+	m1,k=k0 f=6i 9
+	m1,k=k1 f=6i 10
+	m1,k=k0 f=7i 11
+	m1,k=k0 f=5i 12
+	m1,k=k1 f=8i 13
+	m1,k=k2 f=9i 14
+	m1,k=k3 f=5i 15`)
+	assert.NoError(t, err)
+	assert.NoError(t, s.WritePoints(t.Name(), "autogen", models.ConsistencyLevelAny, nil, points))
+
+	u, err := url.Parse(s.URL())
+	assert.NoError(t, err)
+
+	cmd := "connect " + u.Host
+	c := cli.CommandLine{}
+	assert.NoError(t, c.ParseCommand(cmd))
+
+	// Test a successful flux query
+	buf := &bytes.Buffer{}
+	assert.NoError(t, c.ExecuteFluxQuery(buf, `import "influxdata/influxdb/v1" v1.databases()`))
+	assert.Equal(t,
+		`Result: _result
+Table: keys: [organizationID]
+ organizationID:string     databaseName:string  retentionPolicy:string         retentionPeriod:int  default:bool         bucketId:string  
+----------------------  ----------------------  ----------------------  --------------------------  ------------  ----------------------  
+                               TestCliEndToEnd                 autogen                           0          true                          
+`,
+		buf.String())
+
+	// Test a bigger successful flux query
+	buf.Reset()
+	assert.NoError(t, c.ExecuteFluxQuery(buf, `import "influxdata/influxdb/v1"
+v1.databases() |> yield(name: "show_the_dbs")
+from(bucket:"TestCliEndToEnd") |> range(start: 0) |> drop(columns:["_start","_stop"]) |> yield(name: "show_the_data")`))
```

Comment:
Bigger test

Reply:
Should there be any `sort` calls in here to prevent flakiness? Or is it effectively deterministic?

Reply:
It is deterministic - we check against both orderings (`Contains` check below)